http/1.7/modules/ROOT/pages/http-authentication.adoc

```diff
@@ -699,6 +700,227 @@ In the *Configuration XML* editor, the `tls:context`, `tls:trust-store` and `tls
+
+The following example shows how to configure the OAuth2 authorization code grant type authentication for the HTTP *Request* operation by creating a Mule app to access a protected resource and the resource in the resource server. See xref:mule-runtime::authorization-grant-types.adoc[OAuth Authorization Grant Types]. The example covers how to:
+
+* Create the Mule app
+* Add redirection to local authorization URL
+* Add OAuth authentication
+* Add the connected app
+* Add token URL
+* Run the Mule app
+
+[NOTE]
+This example requires you to have access to an Anypoint Platform account with organization administrator permission and be familiar with xref:mule-runtime::configuring-properties.adoc[Configuring Properties].
+
+Before you create the Mule app, review the following steps for getting OAuth access token and returning token for data:
+
+. The user submits an HTTP request in the Mule app.
+. The Mule app forwards the request to the protected resource in the Anypoint organization server.
+. The Anypoint organization server rejects the request because the Mule app doesn't have a token.
+. The Mule app redirects the user to an URL corresponding to the authorization server, indicating where to call the Mule app back when the authorization code is available.
+. The user authenticates itself to the authorization server.
+. The authorization server returns an authorization code back to the redirect URI.
+. The Mule app calls the authorization server again and provides the authorization code to obtain an access token (stored in an object store).
+. The Mule app tries to access the protected resource again, providing a valid access token.
+. The Anypoint organization server validated the token and delivers the protected resource after successful authentication. 
++
+Note that the user is not exposed to the token nor does the Mule app see customer credentials, but just an authorization code. 
+
+=== Create the Mule App
+
+When you create a Mule app, the flow tries to access a protected resource in the Anypoint organization server with no authentication configured:
+
+[source,xml,linenums]
+----
+<http:listener-config name="listenerConfig">
+  <http:listener-connection host="0.0.0.0" port="8070" />
+</http:listener-config>
+
+<http:request-config name="requestConfig">
+  <http:request-connection protocol="HTTPS" host="${host}" port="${port}" />
+</http:request-config>
+
+<flow name="authCodeFlow">
+  <http:listener config-ref="listenerConfig" path="/test" />
+  <http:request method="GET" config-ref="requestConfig" path="${resource}" />
+</flow>
+----
+
+The Mule app consists of an HTTP *Listener* source and an HTTP *Request* operation. If the `host`, `port`, and `resource` properties in your YAML file aren't set to an accessible resource, the Anypoint Platform fails with a 401 status code because it expects to obtain the access token in the request:
```

Comment:
```suggestion
The Mule app consists of an HTTP *Listener* source and an HTTP *Request* operation. If the `host`, `port`, and `resource` properties in your YAML file aren't set to an accessible resource, Anypoint Platform fails with a 401 status code because it expects to obtain the access token in the request:
```

Reply:
Does Anypoint Platform or the request fail? Does "it" refer to AP?

Reply:
It's refering to Anypoint Platform. Can we change the "it" for just "platform" in this scenario?

Reply:
I think "it" is fine if it's Anypoint Platform that fails.